src/client/mod.rs

```diff
@@ -565,6 +565,28 @@ impl Client {
         &self.inner.topology
     }
 
+    pub(crate) async fn primary_description(&self) -> Option<crate::sdam::ServerDescription> {
```

Comment:
The spec includes a clause that creating collections with encrypted fields must be denied if the primary is <7.0.  The reference implementation in the C driver actually does a full connection checkout to test that, but this lets us avoid that.

Unfortunately, simply peeking the latest topology description doesn't work because it's possible to call `create_collection` while the topology is still being discovered, so this waits for data to be populated.